custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,220 @@
+
+    def get_matches(self, video):
+        matches = set()
+        
+        if isinstance(video, Episode):
+            # series name
+            if sanitize(video.series) and sanitize(self.video.series) in (
+                    sanitize(name) for name in [video.series] + video.alternative_series):
+                matches.add('series')
+            
+            # season
+            if video.season and self.video.season is None or video.season and video.season == self.video.season:
+                matches.add('season')            
+
+            # year
+            if video.original_series and self.video.year is None or video.year and video.year == self.video.year:
+                matches.add('year')
+                
+            # episode
+            if video.episode == self.video.episode:
+                matches.add('episode')
+        else:
+            raise ValueError(f"Unhandled instance of argument 'video': {type(video)}")
+
+        return matches
+
+class JimakuProvider(Provider):
+    '''Jimaku Provider.'''
+    video_types = (Episode, Movie)
+    api_url = 'https://jimaku.cc/api'
+    
+    languages = {Language.fromietf("ja")}
+
+    def __init__(self, api_key=None):
+        if api_key:
+            self.api_key = api_key
+        else:
+            raise ConfigurationError('Missing api_key.')
+
+        self.session = None
+
+    def initialize(self):
+        self.session = Session()
+        self.session.headers['User-Agent'] = 'Subliminal/%s' % __short_version__
+        self.session.headers['Content-Type'] = 'application/json'
+        self.session.headers['Authorization'] = self.api_key
+
+    def terminate(self):
+        self.session.close()
+
+    def query(self, video):
+        series_name = str(video.series)
+        episode_number = video.episode
+        
+        # Attempt to derive anilist_id
+        derived_anilist_id = None
+        derive_from_tag = ""
+        
+        # Not yet handled: series_anidb_series_id being '(None, )' (class 'str')
+        for tag in ["series_anidb_id", "tvdb_id", "series_tvdb_id"]:
+            candidate = getattr(video, tag, None)
+            if candidate:
+                derive_from_tag = tag
+                break
+            
+        if derive_from_tag:
+            logger.info(f"Will attempt to derive Anilist ID based on '{derive_from_tag}'")
+            
+            # Left: video, right: anime-lists
+            tag_map = {
+                "series_anidb_id": "anidb_id",
+                "tvdb_id": "thetvdb_id",
+                "series_tvdb_id": "thetvdb_id",
+            }
+            mapped_tag = tag_map[derive_from_tag]
+            
+            try:
+                # We won't use self.session as the endpoint doesnt need our API key
+                response = requests.get(
+                    "https://raw.githubusercontent.com/Fribb/anime-lists/master/anime-list-mini.json",
+                    headers={'Content-Type': 'application/json'}
```

Comment:
Fetching this JSON every single episode / movie is too much, the content is always the same, you should extract into a method and cache it.

There is an annotator that can be used to cache.

Reply:
Good call, perhaps something [like this](https://github.com/morpheus65535/bazarr/pull/2505/commits/a0e6513a9432440d72693b18ee669ad492e416b2#diff-3e47ba3e14d0fd939453aaff0c1c60c644d0c45a507635ffbcf63a9ffbea3dd1R223)?